Fix running tests in parallel

jacek-lewandowski opened the pull request:
Fixed the semaphore which was release twice in some cases

Fixed leaving TAR stream unclosed

Added sleep between subsequent retries of starting worker

pedjak commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -1,5 +1,6 @@
+#Mon Jul 08 09:50:10 CEST 2019
```
upgrading gradle here is not really required, hence I would revert this.

pedjak commented on src/main/java/com/pedjak/gradle/plugins/dockerizedtest/DockerizedExecHandle.java:
```diff
@@ -232,7 +232,7 @@ private void setEndStateInfo(ExecHandleState newState, int exitValue, Throwable
         }
 
-        LOGGER.debug("Process '{}' finished with exit value {} (state: {})", displayName, exitValue, newState);
+        LOGGER.warn("Process '{}' finished with exit value {} (state: {})", displayName, exitValue, newState);
```
please revert this to debug level.

pedjak commented on src/main/java/com/pedjak/gradle/plugins/dockerizedtest/DockerizedExecHandle.java:
```diff
@@ -265,7 +265,7 @@ public ExecHandle start() {
 
             while (stateIn(ExecHandleState.STARTING)) {
-                LOGGER.debug("Waiting until process started: {}.", displayName);
+                LOGGER.warn("Waiting until process started: {}.", displayName);
```
please revert to debug.

pedjak commented on src/main/java/com/pedjak/gradle/plugins/dockerizedtest/DockerizedExecHandle.java:
```diff
@@ -280,7 +280,7 @@ public ExecHandle start() {
             }
 
-            LOGGER.info("Successfully started process '{}'", displayName);
+            LOGGER.warn("Successfully started process '{}'", displayName);
```
please revert to info

pedjak commented on src/main/java/com/pedjak/gradle/plugins/dockerizedtest/ForkingTestClassProcessor.java:
```diff
@@ -47,7 +47,7 @@ public void startProcessing(TestResultProcessor resultProcessor) {
         int i = 0;
         RuntimeException exception = null;
-        while (remoteProcessor == null && i < 10)
+        while (remoteProcessor == null && i < 20)
```
did increasing to 20 really help?

jacek-lewandowski replied:
Yes, and i'd really make it retry for even longer - there is no point in stop retrying so fast given we know that the failures in 99% cases are caused by lack of the resource and it will eventually succeed. Retrying the whole test round is much more expensive and will not be that reliable.

Commit:
Fix running tests in parallel

Fixed the semaphore which was release twice in some cases

Fixed leaving TAR stream unclosed

Added sleep between subsequent retries of starting worker
